### PR TITLE
feat: [CDS-77885]: asg base

### PR DESCRIPTION
**Pull request by litvinovstas**
### Summary
Asg improvements:
Added Asg Service User Data, Asg Infra Base Name and provisioner, baseName for Asg Rolling/Canary/BG steps, supporting for multi load balancers.

#### Screenshots
Screenshot 2023-09-29 at 09 10 36
Screenshot 2023-09-29 at 09 10 58
Screenshot 2023-09-29 at 09 11 10

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/components/PipelineSteps/AsgBlueGreenDeployStep/AsgBGLoadBalancers/AsgBlueGreenDeployLoadBalancers.tsx [hunk 1]
```diff
@@ -0,0 +1,575 @@
...
+
+import { AsgBlueGreenDeployStepInitialValues } from '../AsgBlueGreenDeployStep'
+// import type { ElastigroupBGStageSetupData } from './ElastigroupBGStageSetupStepTypes'
+
+import stepCss from '@pipeline/components/PipelineSteps/Steps/Steps.module.scss'
+import css from './AsgBlueGreenDeployLoadBalancers.module.scss'
+
+export const extractPort = (listenerPort: string): string => {
+  const splitString = listenerPort.split(': ')
+  if (splitString.length !== 2) {
```
Comment (sahithikolichala) [inline src/modules/75-cd/components/PipelineSteps/AsgBlueGreenDeployStep/AsgBGLoadBalancers/AsgBlueGreenDeployLoadBalancers.tsx:35]: can you comment here on why only 2?

File: src/modules/75-cd/components/PipelineSteps/AsgBlueGreenDeployStep/AsgBGLoadBalancers/AsgBlueGreenDeployLoadBalancers.tsx [hunk 2]
```diff
@@ -0,0 +1,575 @@
...
+          </>
+        ) : (
+          <Layout.Vertical>
+            {loadBalancerSummaryView(getString('production'), index, 'prodListener', 'prodListenerRuleArn')}
+            {loadBalancerSummaryView(getString('common.stage'), index, 'stageListener', 'stageListenerRuleArn')}
+          </Layout.Vertical>
+        )}
+      </Collapse>
+    </Card>
+  )
```
Comment (sahithikolichala) [inline src/modules/75-cd/components/PipelineSteps/AsgBlueGreenDeployStep/AsgBGLoadBalancers/AsgBlueGreenDeployLoadBalancers.tsx:574]: Why did we add the empty test file 
src/modules/75-cd/components/PipelineSteps/AsgBlueGreenDeployStep/AsgBGLoadBalancers/__tests__/AsgBlueFreenDeployLoadBalancers.tsx?

File: src/modules/75-cd/components/PipelineSteps/AsgInfraSpec/AsgInfraSpecInputForm.tsx [hunk 3]
```diff
@@ -91,6 +129,12 @@ const AsgInfraSpecInputForm = ({
             label={getString('connector')}
             enableConfigureOptions={false}
             placeholder={getString('common.entityPlaceholderText')}
+            onChange={() => {
+              if (CDS_BASIC_ASG) {
```
Comment (sahithikolichala) [inline src/modules/75-cd/components/PipelineSteps/AsgInfraSpec/AsgInfraSpecInputForm.tsx:133]: try moving the logic out of onchange

**Review by sahithikolichala — COMMENTED** [review]
Please add test coverage for the new flows

**Review by srdjan-harness — DISMISSED** [review]

**Review by ishantmehtaharness — APPROVED** [review]
PR Looks Good

**Review by sahithikolichala — APPROVED** [review]

**Review by ishantmehtaharness — APPROVED** [review]

**Review by srdjan-harness — APPROVED** [review]

**Pull request merged**